src/api/spec/controllers/webui/package_controller_spec.rb

```diff
@@ -38,15 +39,74 @@
         </resultlist>')
   end
 
-  describe 'submit_request' do
+  describe "POST #submit_request" do
+    RSpec.shared_examples "a response of a successfull submit request" do
+      it { expect(flash[:notice]).to match("Created .+submit request \\d.+to .+#{target_project}") }
+      it { expect(response).to redirect_to(package_show_path(project: source_project, package: package)) }
+      it { expect(BsRequestActionSubmit.where(target_project: target_project.name, target_package: package.name)).to exist }
+    end
+
+    before do
+      login(user)
+    end
+
+    context "sending a valid submit request" do
+      before do
+        post :submit_request, { project: source_project, package: package, targetproject: target_project }
+      end
+
+      it_should_behave_like "a response of a successfull submit request"
+    end
+
+    context "having whitespaces in parameters" do
+      before do
+        post :submit_request, { project: " #{source_project} ", package: " #{package} ", targetproject: " #{target_project} " }
+      end
+
+      it_should_behave_like "a response of a successfull submit request"
+    end
+
     context 'not successful' do
       before do
-        login(user)
         post :submit_request, { project: source_project, package: source_package, targetproject: target_project.name }
       end
 
       it { expect(flash[:error]).to eq('Unable to submit: The source of package home:tom/my_package is broken') }
-      it { expect(BsRequestActionSubmit.where(target_project: target_project, target_package: source_package).count).to eq(0) }
+      it { expect(response).to redirect_to(package_show_path(project: source_project, package: source_package)) }
+      it { expect(BsRequestActionSubmit.where(target_project: target_project.name, target_package: source_package.name)).not_to exist }
+    end
+
+    context "a submit request that fails due to validation errors" do
```

Comment:
I don't like that. There are to many variations. The tests would become unreadable. Like, we would have to pass the expected flash error, redirect destination, package name, project name and probably more.